TemporaryParser-Tests/TPRBParserTest.class.st

```diff
@@ -168,104 +178,437 @@ TPRBParserTest >> testBlockReturnNode [
-		do: [ :each | 
-			self
-				assert: ((self parserClass parseMethod: each first) equalTo: (self parserClass parseMethod: (each at: 2)) exceptForVariables: #('c'))
-				identicalTo: each last ]
+{ #category : #'garbage tests' }
+TPRBParserTest >> testGarbageBadlyParenthesizedEndOfStatement [
+	"Should propose a closing parenthese before the end of the statement. Should also offer possibility to quickly erase it."
+	self should: [self parserClass parseExpression: '(temp := 1 + 3 .'] raise: SyntaxErrorNotification.
 ]
 
-{ #category : #'tests parsing' }
+{ #category : #'garbage tests' }
+TPRBParserTest >> testGarbageEncouteringEndOfStatement [
+	"Could propose multiple reparations like assignement, period, array..."
+	self should: [self parserClass parseExpression: 'temp 1 + 3'] raise: SyntaxErrorNotification.
+]
+
+{ #category : #'garbage tests' }
+TPRBParserTest >> testGarbageFaultyCascadeGivesMultipleErrorAndNoMassage [
+	| tree |
+	tree := self parseFaultyExpression: 'faulty cascade; 1; node'.
+	self assert: tree isSequence.
+	self assert: tree statements size equals: 2.
+	
+	"The first node is meant to be a cascade but the failure prevent from having a cascade node which means having an error 
+	 node instead. This has some repercutions : their is no cascade message and the name of the parents are wrong."
+	self assert: tree statements first isFaulty.
+	self assert: tree statements first errorMessage equals: 'Message expected'.
+	
+	self assert: tree statements last isFaulty.
+	self assert: tree statements last errorMessage equals: 'End of statement list encountered'.
+	
+	self assert: tree parent printString equals: 'noMethod
+  1; node.
+  1; node'.
+	
+	"The problem is further enlightenned with longer cascades."
+	tree := self parseFaultyExpression: 'faulty cascade; successful; until; not; 1; node'.
+	self assert: tree isSequence.
+	self assert: tree statements size equals: 2.
+	
+	self assert: tree statements first isFaulty.
+	self assert: tree statements first errorMessage equals: 'Message expected'.
+	
+	self assert: tree statements last isFaulty.
+	self assert: tree statements last errorMessage equals: 'End of statement list encountered'.
+	
+	self assert: tree parent printString equals: 'noMethod
+  1; node.
+  1; node'.
+]
+
+{ #category : #'garbage tests' }
+TPRBParserTest >> testGarbageForgottenDotAfterAssignement [
+	| tree child error |
+	self shouldnt: [self parserClass parseExpression: '|temp | temp := 1 + 2 temp + 3'] raise: SyntaxErrorNotification.

```

Comment:
This one is valid syntax. The first `temp` is a variable. The second `temp` is a unary mesage.

Reply:
I know but the temps in not a method recognized by numbers so the AST should propose to the user a solution recognizing patterns.